packages/rest/src/lib/CDN.ts

```diff
@@ -0,0 +1,143 @@
+	 */
+	public discoverySplash(guildID: string, splashHash: string, options?: ImageURLOptions): string {
+		return this.makeURL(`/discovery-splashes/${guildID}/${splashHash}`, options);
+	}
+
+	/**
+	 * Generates an emoji's URL for an emoji.
+	 * @param emojiID The emoji ID
+	 * @param extension The extension of the emoji
+	 */
+	public emoji(emojiID: string, extension?: ImageExtension): string {
+		return this.makeURL(`/emojis/${emojiID}`, { extension });
+	}
+
+	/**
+	 * Generates a group DM icon URL for a group DM.
+	 * @param channelID The group channel ID that has the icon
+	 * @param iconHash The hash provided by Discord for this group DM channel
+	 * @param options Optional options for the icon
+	 */
+	public groupDMIcon(channelID: string, iconHash: string, options?: ImageURLOptions): string {
+		return this.makeURL(`/channel-icons/${channelID}/${iconHash}`, options);
+	}
+
+	/**
+	 * Generates a banner URL for a guild's banner.
+	 * @param guildID The guild ID that has the banner splash
+	 * @param bannerHash The hash provided by Discord for this banner
+	 * @param options Optional options for the banner
+	 */
+	public guildBanner(guildID: string, bannerHash: string, options?: ImageURLOptions): string {
+		return this.makeURL(`/banners/${guildID}/${bannerHash}`, options);
+	}
+
+	/**
+	 * Generates an icon URL for a guild's icon.
+	 * @param guildID The guild ID that has the icon splash
+	 * @param iconHash The hash provided by Discord for this icon
+	 * @param options Optional options for the icon
+	 */
+	public guildIcon(guildID: string, iconHash: string, options?: ImageURLOptions): string {
+		return this.makeURL(`/icons/${guildID}/${iconHash}`, options);
+	}
+
+	/**
+	 * Generates a guild invite splash URL for a guild's invite splash.
+	 * @param guildID The guild ID that has the invite splash
+	 * @param splashHash The hash provided by Discord for this splash
+	 * @param options Optional options for the splash
+	 */
+	public splash(guildID: string, splashHash: string, options?: ImageURLOptions): string {
+		return this.makeURL(`/splashes/${guildID}/${splashHash}`, options);
+	}
+
+	/**
+	 * Generates a team icon URL for a team's icon.
+	 * @param teamID The team ID that has the icon
+	 * @param iconHash The hash provided by Discord for this icon
+	 * @param options Optional options for the icon
+	 */
+	public teamIcon(teamID: string, iconHash: string, options?: ImageURLOptions): string {
+		return this.makeURL(`/team-icons/${teamID}/${iconHash}`, options);
+	}
+
+	/**
+	 * Generates a user avatar URL for a user's avatar.
+	 * @param userID The user ID that has the icon
+	 * @param avatarHash The hash provided by Discord for this avatar
+	 * @param options Optional options for the avatar
+	 */
+	public userAvatar(userID: string, avatarHash: string, { dynamic = false, ...options }: ImageURLOptions = {}): string {
+		if (dynamic && avatarHash.startsWith('a_')) options.extension = 'gif';
```

Comment:
I would propose for consistency's sake to expand all if statements to be easier on the diff and potential future additions.

```suggestion
		if (dynamic && avatarHash.startsWith('a_')) {
			options.extension = 'gif';
		}
```